server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1339,11 +1436,16 @@ public File exportCertificates(
         poolManager.regenerateDirtyEntitlements(consumer);
 
         List<CertificateSerialDto> allCerts = new LinkedList<CertificateSerialDto>();
-        for (EntitlementCertificate cert : entCertService
-            .listForConsumer(consumer)) {
-            allCerts.add(new CertificateSerialDto(cert.getSerial().getId()));
+        for (Long id : entCertService.listEntitlementSerialIds(consumer)) {
+            allCerts.add(new CertificateSerialDto(id));
+        }
+        // add content access cert if needed
+        if (!StringUtils.isEmpty(consumer.getOwner().getContentAccessMode())) {
```

Comment:
These chained calls may end up being performance pain points later, as I believe Hibernate will lazily load the Owner, and then pull the access mode. Nothing to change quite yet, but something to keep in mind for the future.